### PR TITLE
Add Documentation in `Autodesk.Forge.Core`

### PR DESCRIPTION
**Pull request by ricaun**
Adding missing documentation in the `Autodesk.Forge.Core` project.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <Project>
  <PropertyGroup>
+   <Version>4.0.1</Version>
```
Comment (szilvaa) [inline Directory.Build.props:3]: Couldn't we just define `<PackageVersion>4.0.1</PackageVersion>` here? Why do we need the `Version` variable?

**Review by szilvaa — COMMENTED** [review]
Thanks for opening this PR.

File: Directory.Build.props (hunk 1 above)
Comment (ricaun) [reply Directory.Build.props:3]: The `Version` is for the assembly and the `PackageVersion` is for the package.

Kinda forget that you can remove the `PackageVersion` and the `Version` gonna be used in the package version generation.

Comment (ricaun) [reply Directory.Build.props:3]: > Couldn't we just define `<PackageVersion>4.0.1</PackageVersion>` here? Why do we need the `Version` variable?

Yes you could use `PackageVersion` in there, the `Version` gonna apply in the assembly file as well.

Comment (ricaun) [reply Directory.Build.props:3]: Without `Version` the assembly gonna have version `1.0.0` and the package a different number.

**Comment by ricaun** [comment]
Kinda have a question why the `TargetFramework` changed to `net8.0`? 

The projects still build with `net6.0`, I don't understand the reason, I still can use the `net6.0` package in a `net8.0` project.

Using `<TargetFrameworks>net6.0;net8.0</TargetFrameworks>` would build in both versions. 😄

**Comment by szilvaa** [comment]
> Kinda have a question why the `TargetFramework` changed to `net8.0`?

.net6 will be out of support in a few months.

**Review by szilvaa — APPROVED** [review]

**Comment by ricaun** [comment]
> > Kinda have a question why the `TargetFramework` changed to `net8.0`?
> 
> .net6 will be out of support in a few months.

Usually packages are built on the lowest version to give more flexibility to the user, I was wondering if this package should be `netstandard2.0` to join the main ASP SDK ecosystem. I think this will never happen.

Anyway, could you please create a Release in this project to match the NuGet release. By releasing in the GitHub everyone that like this project or interact in some way gonna be know that the version 4.0.1 is released.

**Comment by szilvaa** [comment]
> netstandard2.0 to join the main ASP SDK ecosystem. I think this will never happen.

Yes, there's no plan to support netstandard2.0 again. Note that we used to, but it lacked some APIs that we wanted to use so we moved to .net(core). I expect that moving forward this package will upgrade to .net LTS releases like Autodesk desktop products.

**Comment by ricaun** [comment]
> Yes, there's no plan to support netstandard2.0 again. Note that we used to, but it lacked some APIs that we wanted to use so we moved to .net(core). I expect that moving forward this package will upgrade to .net LTS releases like Autodesk desktop products.

Yep, `netstandard2.0` is kinda complicated I tried but to many strong dependencies with net code.

About supporting net6.0 and net8.0 is kinda simple to do it, I gonna create a PR.

**Pull request merged**